Source/missiles.cpp

```diff
@@ -2283,12 +2283,16 @@ void AddFlamec(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy,
 
 void AddCbolt(int mi, int sx, int sy, int dx, int dy, int midir, char micaster, int id, int dam)
 {
+	/// ASSERT: assert((DWORD)mi < MAXMISSILES);
+
 	if (micaster == 0) {
-		if (id != -1)
+		if (id == myplr) {
 			missile[mi]._mirnd = random(63, 15) + 1;
-		else
+			missile[mi]._midam = random(68, plr[id]._pMagic >> 2) + 1;
+		} else {
 			missile[mi]._mirnd = random(63, 15) + 1;
-		missile[mi]._midam = random(68, plr[id]._pMagic >> 2) + 1;
+			missile[mi]._midam = random(68, plr[id]._pMagic >> 2) + 1;
+		}

```

Comment:
Why duplicate this line back into the both conditional branches as it was done for the shorter random() line?

Reply:
because this way it gets binexact, otherwise it doesn't : P

Reply:
To be fair if jump was optimized away it's unclear if `_midam` assignment was done in both branches or not. Maybe there were some other evidence of this though, I'm not sure.

Reply:
We can infer from the logic of several other functions what the intention was here. `MI_Lightctrl` is a good example. Given that we know it checks `id == myplr`, and the damage statement uses `plr[id]._pMagic`. It seems possible they may have intended to use player's magic value if the player is casting, or else something else if it's another player/missile.